src/unstract/adapters/embedding/azure_open_ai/src/azure_open_ai.py

```diff
@@ -51,12 +51,12 @@ def get_json_schema() -> str:
         f.close()
         return schema
 
-    def get_embedding_instance(self) -> Optional[BaseEmbedding]:
+    def get_embedding_instance(self) -> BaseEmbedding:
```

Comment:
@gaya3-zipstack for all the adapter's `get_*()` methods that throw an error, can you make use of [these specific error classes](https://github.com/Zipstack/unstract-adapters/blob/0c55bf8079c8026632cb305274d8aa854a23302a/src/unstract/adapters/exceptions.py#L14-L27) for the same?

Reply:
this need one more refactoring. Will pick it up later